### PR TITLE
Also look at body.code === system.error when rejecting an API call EXP-4017

### PR DESCRIPTION
**Pull request by brianfeister**
…XP-4017

### CODE REVIEW

**Comment by brianfeister** [comment]
Added @jrd261 just to be sure it doesn't get lost in a storm of meetings 😂

**Comment by keck** [comment]
sorry :)  I literally started my day with popping this up, read it, code looks fine, wanted to test it, built CLI, then got pulled into the day.

**Comment by brianfeister** [comment]
@keck, no worries at all. Just keeping the conversation going.

@jrd261 I'll need a specific code snippet where you see a problem. My understanding is that the `request.post` second argument is a callback and that `err === null` if the request doesn't convey a `4xx` error.

My understanding of the problem is that since the JCR was throwing an internal service error, either JCR or API was not setting http status (at least it didn't set `4xx` in my testing) and so I'm just adding one layer to look for `system.error` in `body.code` since that's the only place where an error was communicated when JCR was in this bad state.

```javascript
 [...]
      }, (err, response, body) => {
        if (err || body.code === 'system.error' ) return reject(err || body);

        resolve(JSON.parse(body));
      });
  })
```

**Comment by jrd261** [comment]
Sorry for the delay. Something like this:

```javascript
request(options, (error, response, body) => {
  if (error) {
    reject({ code: 'network.error', message: error.toString() });
  } else if (response.statusCode >= 400) {
    if (response.headers['content-type'] === 'application/json') reject(JSON.parse(response.body));
    else reject({ code: 'unknown.error', message: response.body.toString() });
  } else {
    resolve(JSON.parse(response.body));
  }
});
```

I'm pretty sure "error" has nothing to do with http status code. The request is an "error" only if you can't connect to the backend. The request can be successful and receive a 500. Its up to checking against the status code to handle "successful" requests with status codes that indicate the operation failed.

**Comment by brianfeister** [comment]
@jrd261 Gotcha. Ready for review again, made those changes.

File: lib/deploy/upload.js [hunk 1]
```diff
@@ -24,12 +24,21 @@ function upload(exp, data) {
...
+            reject({ code: 'unknown.error', message: body.toString() });
+          }
+        } else {
+          resolve(JSON.parse(body));
+        }
       });
   })
-  .then(res => exp.getContent(res.uuid));
+  .then(res => exp.getContent(res.uuid))
+  .catch( err => err );
```
Comment (brianfeister) [inline lib/deploy/upload.js]: @jrd261 the various upload methods below all call this method but the `.catch` treats errors differently with different messages, so "yes", the consumer of this `throw`s the error and logs a message to the console that's different depending on context.

**Comment by jrd261** [comment]
:pizza:

**Pull request merged**